clang/test/SemaSYCL/reqd-sub-group-size-host.cpp

```diff
@@ -1,9 +1,53 @@
-// RUN: %clang_cc1 -fsycl -fsycl-is-host -fsyntax-only -verify %s
-// expected-no-diagnostics
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -Wno-sycl-2017-compat -fsyntax-only -verify %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -Wno-sycl-2017-compat -ast-dump %s | FileCheck %s
+//expected-no-diagnostics
 
 [[intel::reqd_sub_group_size(8)]] void fun() {}
 
 class Functor {
 public:
-  [[intel::reqd_sub_group_size(16)]] void operator()() {}
+  [[intel::reqd_sub_group_size(16)]] void operator()() const {}
 };
+
+class Functor16 {
+public:
+  [[intel::reqd_sub_group_size(4)]] void operator()() const {
+  }
+};
+
+class Functor4 {
+public:
+  [[intel::reqd_sub_group_size(12)]] void operator()() const {}
+};
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
+  kernelFunc();
+}
```

Comment:
Same here about single style.

Reply:
test is not updated since mock header does not work on host.